### PR TITLE
Tutorial for WS-Federation middleware

### LINKED ISSUE #4523: New docs for WS-Federation auth
1. Write a **short** paragraph describing what the topic will cover.
WsFederation is a new authentication handler that has been ported from Microsoft.Owin / ASP.NET 4.5. It is being released out of band against ASP.NET Core 2.0.0 dependencies. See https://github.com/aspnet/Security/issues/1473. 

2. Write an outline of the proposed topic contents.
There should be a walk through similar to these: https://docs.microsoft.com/en-us/aspnet/core/security/authentication/social/

3. Where would you put the topic in the Table of Contents.
https://docs.microsoft.com/en-us/aspnet/core/security/authentication/

### PR DESCRIPTION
**Pull request by chlowell**
Fixes #4523 

@scottaddie @Tratcher here's a short walkthrough of authenticating with ADFS via WS-Fed. It would be nice to link to documentation of `WsFederationOptions`. Does that exist anywhere?

[Internal Review Page](https://review.docs.microsoft.com/en-us/aspnet/core/security/authentication/ws-federation?branch=pr-en-us-5532)

### CODE REVIEW

**Comment by scottaddie** [comment]
@Tratcher Would you mind reviewing this one too?

**Comment by Rick-Anderson** [comment]
@chlowell  outstanding! Can you check [my minor copy-edits](https://github.com/aspnet/Docs/pull/5532/commits/666d5b8320a4444718509fb6b299a21ba2018f68)?

**Comment by chlowell** [comment]
They look good to me, thanks for the review.

**Comment by Rick-Anderson** [comment]
@Tratcher  can you do a quick review. This is short and well written.

**Comment by Tratcher** [comment]
Nice, thanks @chlowell. I wouldn't consider #4523 resolved until we added a section for AAD as well, but that can be added in a subsequent PR.

File: aspnetcore/security/authentication/ws-federation.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+For ASP.NET Core 2.0 apps, WS-Federation support is provided by [Microsoft.AspNetCore.Authentication.WsFederation](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.WsFederation). This component is ported from [Microsoft.Owin.Security.WsFederation](https://www.nuget.org/packages/Microsoft.Owin.Security.WsFederation) and shares many of that component's mechanics. However, the components differ in a couple of important ways.
+
+By default, the new middleware:
+
+* Doesn't allow unsolicited logins. This feature of the WS-Federation protocol is vulnerable to XSRF attacks. However, it can be enabled with the `AllowUnsolicitedLogins` option.
+* Doesn't check every form post for sign-in messages. Only requests to the `CallbackPath` are checked for sign-ins. `CallbackPath` defaults to `/signin-wsfed` but can be changed. This path can be shared with other authentication providers by enabling the `SkipUnrecognizedRequests` option.
+
+## Use ADFS as an external login provider
```
Comment (Tratcher) [inline aspnetcore/security/authentication/ws-federation.md]: for ASP.NET Core Identity

**Comment by chlowell** [comment]
A section on AAD would be short and simple, I can add one to this PR.

**Comment by scottaddie** [comment]
@Tratcher Please review the new AAD section.

**Comment by scottaddie** [comment]
@chlowell Thank you again for the fantastic work on this doc! We truly appreciate you taking the time to do this.

**Pull request merged**